internal/ignore/ignore.go

```diff
@@ -0,0 +1,129 @@
+package ignore
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	domainutils "github.com/supergoodsystems/supergood-go/internal/domain-utils"
+	remoteconfig "github.com/supergoodsystems/supergood-go/internal/remote-config"
+	"github.com/supergoodsystems/supergood-go/internal/shared"
+)
+
+// ShouldIgnoreRequest evaluates based off the remote config cache whether an intercepted
+// request should be forwarded to supergood
+func ShouldIgnoreRequest(req *http.Request, rc *remoteconfig.RemoteConfig) (bool, []error) {
+	var errs []error
+	domain := domainutils.GetDomainFromHost(req.Host)
+	if domain == "" {
+		return false, errs
+	}
+
+	endpoints := rc.Get(domain)
+	if len(endpoints) == 0 {
+		return false, errs
+	}
+	for _, endpoint := range endpoints {
+		if endpoint.Action != "Ignore" {
+			continue
+		}
+		testVal, err := stringifyRequestAtLocation(req, endpoint.Location)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		testByteArray := []byte(fmt.Sprintf("%v", testVal))
+		match := endpoint.Regex.Match(testByteArray)
+		if match {
+			return true, errs
+		}
+	}
+	return false, errs
+}
+
+// stringifyRequestAtLocation takes an endpoint location, which is used to uniquely classify
+// a request and stringifies the request object at that location
+func stringifyRequestAtLocation(req *http.Request, location string) (string, error) {
+	if location == shared.URLStr {
+		return req.URL.String(), nil
+	}
+	if location == shared.DomainStr {
+		return domainutils.Domain(req.URL.String()), nil
+	}
+	if location == shared.SubDomainStr {
+		return domainutils.Subdomain(req.URL.String()), nil
+	}
+	if location == shared.PathStr {
+		return req.URL.Path, nil
+	}
+	if strings.Contains(location, shared.RequestHeadersStr) {
+		return getHeaderValueAtLocation(req.Header, location)
+	}
+	if strings.Contains(location, shared.RequestBodyStr) {
+		return getRequestBodyValueAtLocation(req, location)
+	}
+
+	return "", fmt.Errorf("unexpected location parameter for RegExp matching: %s", location)
+}
+
+// getRequestBodyValueAtLocation retrieves the nested field value of a struct based on the given "location"
+func getRequestBodyValueAtLocation(req *http.Request, location string) (string, error) {
```

Comment:
does this handle arrays as part of the key map? I tried to piece through it but I couldn't see the spot where it might

Reply:
when we originally implemented, I dont think we said we would provide support for ignoring/accepting requests based off nested array fields in the request body. Can look to include if we decide.